src/core/control/xojfile/LoadHandler.cpp

```diff
@@ -902,6 +902,10 @@ void LoadHandler::parserEndElement(GMarkupParseContext* context, const gchar* el
     if (handler->pos == PARSER_POS_STARTED && strcmp(elementName, handler->endRootTag) == 0) {
         handler->pos = PASER_POS_FINISHED;
     } else if (handler->pos == PARSER_POS_IN_PAGE && strcmp(elementName, "page") == 0) {
+        // handle unnecessary layer insertion in case of existing layers in file
+        if (handler->page->getLayerCount() > 1) {
+            handler->page->removeLayer(handler->page->getLayers()->at(0));
+        }
```

Comment:
I find this code block a little weird. I think it'd be better if:

- The Page constructor had a flag "suppressLayerCreation" (defaulting to false), so that the loader would not create an empty layer and then remove it (in most cases).
- Then when a page is finished parsing, and if there is no layers in it, add an empty layer.

Reply:
Sure, I can change that logic.

Reply:
One way around this would be to only contruct the page instance once the parser is done parsing its content (the layers). This could be done in a later PR if someone thinks it matters enough.